src/container.cpp

```diff
@@ -693,12 +692,13 @@ void Container::postAddNotification(Thing* thing, const Cylinder* oldParent, int
 	}
 }
 
-void Container::postRemoveNotification(Thing* thing, const Cylinder* newParent, int32_t index, cylinderlink_t)
+void Container::postRemoveNotification(std::shared_ptr<Thing> thing, std::shared_ptr<const Thing> newParent,
+                                       int32_t index, cylinderlink_t)
 {
-	Cylinder* topParent = getTopParent();
+	auto topParent = getTopParent();
```

Comment:
Could this be `const auto`
```suggestion
	const auto topParent = getTopParent();
```